docs/design/interceptors.md

```diff
@@ -364,7 +364,12 @@ public interface Interceptor<
 /**
  * [Interceptor] context used for all phases that only have access to the operation input (request)
  */
-public interface RequestInterceptorContext<I> : Attributes {
+public interface RequestInterceptorContext<I> {
+  
+  /**
+   * The [ExecutionContext] used to drive the execution of a single request/response
+   */
+  public val executionContext: ExecutionContext
```

Comment:
**Nit**: Please add a link to this design from [**docs/design/README.md**](https://github.com/awslabs/smithy-kotlin/blob/main/docs/design/README.md).